smt/problems/problem.py

```diff
@@ -83,7 +83,12 @@ def sample(self, n):
         x, _ = self.design_space.sample_valid_x(n)
         return x
 
-    def __call__(self, x: np.ndarray, kx: Optional[int] = None) -> np.ndarray:
+    def __call__(
+        self,
+        x: np.ndarray,
+        delta: Optional[np.ndarray] = None,
```

Comment:
Why `delta` name while we pass `self.eval_is_acting` below. I would also prefer to add this new optional parameter at the end (it won't break legacy code).